explorer/interpreter/type_checker.cpp

```diff
@@ -1720,9 +1719,9 @@ auto TypeChecker::MatchImpl(const InterfaceType& iface,
 }
 
 auto TypeChecker::MakeConstraintWitness(
-    const ConstraintType& constraint,
+    const ConstraintType& /*constraint*/,
```

Comment:
I've sent you #2311 to remove these unused parameters and the one in `DeduceCallBindings`.